metadata_test.go

```diff
@@ -0,0 +1,84 @@
+package sarama
+
+import (
+	"sync"
+	"testing"
+	"time"
+
+	"github.com/stretchr/testify/require"
+)
+
+func TestMetadataRefresh(t *testing.T) {
+	var called = make(map[string]int)
+	refresh := newSingleFlightRefresher(func(topics []string) error {
+		time.Sleep(100 * time.Millisecond)
+		// There should never be two refreshes that run concurrently,
+		// so it's safe to change the map with no lock.
+		for _, topic := range topics {
+			called[topic]++
+		}
+		return nil
+	})
+	go func() {
+		require.NoError(t, refresh([]string{"topic1"}))
+	}()
```

Comment:
From the other test file, you cannot call `t.FailNow` on a goroutine that is not the goroutine running the test.